### DISCUSSION TITLE
Is there a way to separate my templates inside of `client/templates` by app?

### DISCUSSION THREAD

**Discussion by mmonj** [silviogutierrez/reactivated#275] [Q&A]
Rather than have all of my templates be direct children of `client/templates/`, I'd like to separate them as indicated below and still have a way to create python templates for these:

```
client
│   index.tsx
│
└───templates
    ├───anotherApp
    │       AnotherAppIndex.tsx
    │       SomeOtherPage.tsx
    │
    └───app
            AppIndex.tsx
            SomePage.tsx
```

If there is currently no way to do so, perhaps a good way to implement it would be to be able to declare an `app_name="myappname"` inside of `templates.py` in the same way that it can be declared under the app's `urls.py`

Thanks in advance

**Comment by silviogutierrez** [comment]
So, this is actually how it *should* be. I like app separation in general too. And if you look here: https://github.com/silviogutierrez/reactivated/blob/db5f01ca843c742357d48461be1c2fedaff1869a/reactivated/pick.py#L230

you'll see that models are already exported by apps. Try it: use `Pick` on a model and `import {models} from "@reactivated"` will have them properly split up by app/file.

The problem is that there's no real canonical way to find the app name from a file, especially if you don't just use APP_NAME/templates.py. Unless we *force* templates to be in `templates.py`, it could get messy.

I'm open to the idea of shifting to this, and would also welcome any PRs.

**Comment by mmonj** [comment]
I suppose using the namespaced `app_name` wouldnt be ideal since that could be set to an entirely different name than the actual app?

**Reply by silviogutierrez** [reply]
I think maybe we can just use paths relative to the BASE_DIR or something like that. What's your project structure like, so I have more sample data?
